DNN Platform/Library/Common/Lists/ListController.cs

```diff
@@ -1,416 +1,627 @@
-                }
-            }
-
-            return list;
-        }
-
-        private Dictionary<string, ListInfo> FillListInfoDictionary(IDataReader dr)
-        {
-            var dic = new Dictionary<string, ListInfo>();
-            try
-            {
-                while (dr.Read())
-                {
-                    // fill business object
-                    ListInfo list = this.FillListInfo(dr, false);
-                    if (!dic.ContainsKey(list.Key))
-                    {
-                        dic.Add(list.Key, list);
-                    }
-                }
-            }
-            catch (Exception exc)
-            {
-                Exceptions.LogException(exc);
-            }
-            finally
-            {
-                // close datareader
-                CBO.CloseDataReader(dr, true);
-            }
-
-            return dic;
-        }
-
-        private Dictionary<string, ListInfo> GetListInfoDictionary(int portalId)
-        {
-            string cacheKey = string.Format(DataCache.ListsCacheKey, portalId);
-            return CBO.GetCachedObject<Dictionary<string, ListInfo>>(
-                new CacheItemArgs(
-                cacheKey,
-                DataCache.ListsCacheTimeOut,
-                DataCache.ListsCachePriority),
-                c => this.FillListInfoDictionary(DataProvider.Instance().GetLists(portalId)));
-        }
-
-        private IEnumerable<ListEntryInfo> GetListEntries(string listName, int portalId)
-        {
-            string cacheKey = string.Format(DataCache.ListEntriesCacheKey, portalId, listName);
-            return CBO.GetCachedObject<IEnumerable<ListEntryInfo>>(
-                new CacheItemArgs(
-                cacheKey,
-                DataCache.ListsCacheTimeOut,
-                DataCache.ListsCachePriority),
-                c => CBO.FillCollection<ListEntryInfo>(DataProvider.Instance().GetListEntriesByListName(listName, string.Empty, portalId)));
-        }
-    }
-}
+    /// <summary>
+    /// Provides access to Dnn Lists.
+    /// </summary>
+    public class ListController
+    {
+        /// <summary>
+        /// The list of list types that are not localized.
+        /// </summary>
+        [Obsolete("Deprecated in v9.8.1, use UnLocalizedLists instead, schedule removal in v11.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "StyleCop.CSharp.MaintainabilityRules",
+            "SA1401:Fields should be private",
+            Justification = "Make private in v11.")]
+        public readonly string[] NonLocalizedLists = UnLocalizableLists;
+
+        private static readonly string[] UnLocalizableLists = { "ContentTypes", "Processor", "DataType", "ProfanityFilter", "BannedPasswords" };
+
+        /// <summary>
+        /// Gets the lists that do not support localization.
+        /// </summary>
+        public string[] UnLocalizedLists => UnLocalizableLists;

```

Comment:
Per [CA1819](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1819), properties shouldn't return arrays (because they would allow editing the underlying data).
```suggestion
        public IReadOnlyCollection<string> UnLocalizedLists => UnLocalizableLists;
```